tests/cacl/test_cacl_application.py

```diff
@@ -153,6 +153,115 @@ def clean_scale_rules(duthosts, enum_rand_one_per_hwsku_hostname, collect_ignore
+                    "forwarding-action": action
+                }
+            },
+            "config": {
+                "sequence-id": index
+            },
+            "ip": {
+                "config": {
+                    "source-ip-address": src_ip
+                }
+            }
+        }
+
+        snmp_acl_entry.update(acl_entry)
+        ssh_acl_entry.update(acl_entry)
+        ntp_acl_entry.update(acl_entry)
+
+    rules_data['acl'] = {
+        "acl-sets": {
+            "acl-set": {
+                "SNMP-ACL": {
+                    "acl-entries": {
+                        "acl-entry": snmp_acl_entry
+                    },
+                    "config": {
+                        "name": "SNMP-ACL"
+                    }
+                },
+                "ssh-only": {
+                    "acl-entries": {
+                        "acl-entry": ssh_acl_entry
+                    },
+                    "config": {
+                        "name": "ssh-only"
+                    }
+                },
+                "ntp-acl": {
+                    "acl-entries": {
+                        "acl-entry": ntp_acl_entry
+                    },
+                    "config": {
+                        "name": "ntp-acl"
+                    }
+                }
+            }
+        }
+    }
+
+    duthost.copy(content=json.dumps(rules_data, indent=4), dest=file_path)
+
+    cmds = 'acl-loader update full {}'.format(file_path)
+    duthost.command(cmds)
+
+    logger.info('Waiting for all rules to be applied...')
+    # "acl-loader update full **.json" command will refresh iptables, we have to
+    # add the ACCEPT SSH iptables rule after acl-loader command. But on multi-asic
+    # testbed, it always costs minutes to sync iptables rules after updating cacl
+    # rules, if sleep for more than 3 mins with time.sleep, then, the process tries to
+    # add the ACCEPT SSH rule, at this point, SSH connection is disconnected now
+    # because the default SSH timeout is 30s, next duthost.command will try to reconnect
+    # to DUT, it will trigger ssh login, it will be rejected by default CACL DENY rule,
+    # test will fail. We have wait_until to solve this problem, here add wail_until
+    # to call check_iptable_rules every 10s to keep ssh session alive, it just calls
+    # duthost.command to active ssh connection.
+    # In this way, we can active ssh connection and wait as long as we want.
+
+    # It has to wait cacl rules to be effective.
+    wait_until(200, 10, 2, check_iptable_rules, duthost)
+    # add ACCEPT rule for SSH to make sure testbed access
+    duthost.command("iptables -I INPUT 3 -p tcp -m tcp --dport 22 -j ACCEPT")
+
+    yield file_path
+
+    logger.info(f"Deleting {file_path}...")
+    duthost.file(path=file_path, state='absent')
+
+    logger.info("Reloading config to recover original ACL configuration...")
+    config_reload(duthost, safe_reload=True, check_intf_up_ports=True)
```

Comment:
Can we revert the acl table/rule by 'config' commands, rather than config_reload?
In test acl, the function will be duplicated for quite a lot times, hence actually lots of config_reload will be executed.

Reply:
I think it might be a little difficult in this situation, because I'm applying about ~63 dummy rules across 3 tables using `acl-loader update full`, so I'd probably have to loop over the list of actual acl rules and run the delete command them one at a time, which would probably take a while, what do you think?

Reply:
Maybe GCU would be a good choice in this situation, it can set a checkpoint and rollback to the checkpoint, sample: https://github.com/sonic-net/sonic-mgmt/blob/6b38d75257e3925d8b91c3820938b0896bf85723/tests/dhcp_relay/test_dhcp_relay.py#L102
But as far as I know, it doesn't support multi-asic devices, (@wen587 FYI)
so we're good with the config reload for now, we can consider refine in the future.